### PR TITLE
Fix: APP-2493

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- Add React Query to finance fetches

Task: [APP-2493](https://aragonassociation.atlassian.net/browse/APP-2493)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [x] I have tested my code on the test network.

[APP-2493]: https://aragonassociation.atlassian.net/browse/APP-2493?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/services/token/token-service.ts [hunk 1]
```diff
@@ -98,37 +99,46 @@ class TokenService {
...
+                },
+              };
+            });
+        });
       },
-    };
+    });
   };
 
   private fetchCoingeckoToken = async (
```
Comment (Fabricevladimir) [inline src/services/token/token-service.ts]: This function and the others in this service are merely the fetchers we use inside of our react-query queries. You can see it being used in `use-token.ts`.

File: src/services/token/token-service.ts [hunk 2]
```diff
@@ -98,37 +99,46 @@ class TokenService {
...
-        year: 0,
+    return queryClient.fetchQuery({
+      queryKey: ['fetchCovalentToken', url],
+      staleTime: 1000 * 60 * 60 * 24, // 24 hours
+      queryFn: () => {
+        return fetch(url, {headers}).then(res => {
+          return res
+            .json()
+            .then((parsed: CovalentResponse<CovalentToken[]>) => {
+              const data = parsed.data?.[0];
```
Comment (jamesej) [inline src/services/token/token-service.ts]: Nicer would be
```
  .then(res => res.json())
  .then((parsed: CovalentResponse<CovalentToken[]>) => {
```
Minor though, doesn't really need a change

File: src/services/token/token-service.ts [hunk 3]
```diff
@@ -151,28 +161,39 @@ class TokenService {
...
-        month: data.market_data.price_change_percentage_30d_in_currency.usd,
-        year: data.market_data.price_change_percentage_1y_in_currency.usd,
+    return queryClient.fetchQuery({
+      queryKey: ['fetchCoingeckoToken', url],
+      staleTime: 1000 * 60 * 60 * 24, // 24 hours
+      queryFn: () => {
+        return fetch(url)
+          .then(res => {
+            return res.json().then((data: CoingeckoToken) => {
+              return {
```
Comment (jamesej) [inline src/services/token/token-service.ts]: Same thing with promise chaining as before

File: src/services/token/token-service.ts [hunk 4]
```diff
@@ -333,9 +354,16 @@ class TokenService {
...
     };
 
-    const res = await fetch(url, options);
-    const parsed = await res.json();
-    const transfers: AlchemyTransfer[] = parsed?.result?.transfers || [];
+    const transfers: AlchemyTransfer[] = await queryClient.fetchQuery({
+      queryKey: ['fetchAlchemyErc20Transfers', walletAddress],
+      queryFn: () => {
+        return fetch(url, options).then(res => {
+          return res.json().then(parsed => {
```
Comment (jamesej) [inline src/services/token/token-service.ts]: Promise chaining again

**Review by jamesej — APPROVED** [review]
All good and working

**Pull request merged**